doc/locale_api.md

```diff
@@ -42,20 +42,123 @@ org.freedesktop.locale1 interface -     -
 
 - Just use the systemd API, don't add any API of our own
 
-#### Design of Proposal Layer (uppper)
+#### Design of Proposal Layer (upper)
 
 - when setting the locale, adjust the proposed package selection and keyboard
   accordingly. And timezone.
 
-Agama.locale1 interface (on Agama/Language1?)
-- is a copy of freedesktop.locale1,
-  forwarding to the elementary layer,
-  and SetLocale does the additional work
+
+
+The general design of the proposal layer is
+
+- declarative, using read-write properties
+- setting some properties will make changes in the proposal layer of other
+  properties of other objects
+
+So here, setting `Locale` below will set also `X11Keyboard` here and
   - Agama...Software...todo(...)
-  - Agama.locale1.SetVConsoleKeyboard(...)
-  - Agama.locale1.SetX11Keyboard(...)
   - Agama...Timezone...todo(...)
 
+For the first version of the API, let's keep things simple:
+
+**LocaleType** is just one string, the value for the `LANG` variable, like
+`"cs_CZ.UTF-8"`.
+
+**X11KeyboardType** is a pair of strings, the Layout and Variant, for example
+`("cz", "qwerty")` or `("us", "basic")`.
+
+We don't expose the console keyboard, instead letting systemd do it via the
+_convert_ parameter.
+
+(The other systemd keyboard settings are X11Model and X11Options, we don't
+have UI or data for that)
+
+```
+# this is gdbus syntax BTW
+node ...Agama/Locale1 {
+  interface ...Agama.Locale1 {
+    methods:
+      # FIXME: the method part is unclear to me. anyone better at proposals?
+      Commit();
```

Comment:
I think we also need proposal method to 1. do initial proposal for given product 2. allow to reset values.
Commit method is fine for me.